File: artifacts/tulipindicators/decompose/src/typprice.c
int ti_typprice_start(const double *options)
{
  (void) options;
  return 0;
}


/*** DEPENDENCIES:
***/


int ti_typprice(int size, const double * const *inputs, const double *options, double * const *outputs)
{
  const double *high = inputs[0];
  const double *low = inputs[1];
  const double *close = inputs[2];
  (void) options;
  double *output = outputs[0];
  int i;
  for (i = 0; i < size; i += 1)
  {
    output[i] = ((high[i] + low[i]) + close[i]) * (1.0 / 3.0);
  }

  return 0;
}


/*** DEPENDENCIES:
***/

File: artifacts/tulipindicators/translate_ablation_1/src/typprice.rs
use crate::*;

pub fn ti_typprice_start(options: Option<&f64>) -> i32 {
    let _ = options; // Explicitly ignore the parameter
    0
}
pub fn ti_typprice(size: i32, inputs: &[&[f64]], options: &[f64], outputs: &mut [&mut [f64]]) -> i32 {
    let high = inputs[0];
    let low = inputs[1];
    let close = inputs[2];
    let output = &mut outputs[0];
    
    for i in 0..size as usize {
        output[i] = ((high[i] + low[i]) + close[i]) * (1.0 / 3.0);
    }
    
    0
}
